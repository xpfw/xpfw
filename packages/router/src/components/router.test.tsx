import * as render from "preact-render-to-json"
import * as React from "react"
import Router from "../components/router"
import RouterStore from "../store/router"

const makeMockElement = (name: string) => {
  return class extends React.Component<any, any> {
    public render() {
      return (
        <div className={name}>
          {JSON.stringify(this.props)}
        </div>
      )
    }
  }
}

test("Basic Fields Test", () => {
  const emptycomp = makeMockElement("404")
  expect(render(<Router emptyComponent={emptycomp} />)).toMatchSnapshot("renderig 404")
  RouterStore.registerRoute("/", makeMockElement("home"))
  RouterStore.registerRoute("/a", makeMockElement("a"))
  RouterStore.registerRoute("/b", makeMockElement("b"))
  RouterStore.registerRoute("/c", makeMockElement("c"))
  RouterStore.visit({path: "/b", params: {my: "arg"}})
  expect(render(<Router emptyComponent={emptycomp} />)).toMatchSnapshot("renderig b")
  RouterStore.visit({path: "/c", params: {gfsdf: "tas"}})
  expect(render(<Router emptyComponent={emptycomp} />)).toMatchSnapshot("renderig c")
  RouterStore.visit({path: "/a", params: {xvcb: "asdf"}})
  expect(render(<Router emptyComponent={emptycomp} />)).toMatchSnapshot("renderig a")
  RouterStore.visit({path: "/a", params: {gewar: "yvcxv"}})
  expect(render(<Router emptyComponent={emptycomp} />)).toMatchSnapshot("renderig a different params")
  RouterStore.back()
  expect(render(<Router emptyComponent={emptycomp} />)).toMatchSnapshot("renderig a with old params because back")
  RouterStore.back()
  expect(render(<Router emptyComponent={emptycomp} />)).toMatchSnapshot("back to c")
  RouterStore.back()
  expect(render(<Router emptyComponent={emptycomp} />)).toMatchSnapshot("back to b")
  RouterStore.back()
  expect(render(<Router emptyComponent={emptycomp} />)).toMatchSnapshot("back to home")
  RouterStore.back()
  RouterStore.back()
  RouterStore.back()
  expect(render(<Router emptyComponent={emptycomp} />)).toMatchSnapshot("back to home multi back no effect")
  RouterStore.visit({path: "/gfadsdf", params: {iou: ".-รถ"}})
  RouterStore.visit({path: "/gfadsdf", params: {jk: "hj"}})
  expect(render(<Router emptyComponent={emptycomp} />)).toMatchSnapshot("invalid oute")
  RouterStore.back()
  expect(render(<Router emptyComponent={emptycomp} />)).toMatchSnapshot("still invalid oute")
  RouterStore.back()
  expect(render(<Router emptyComponent={emptycomp} />)).toMatchSnapshot("back home")
})
