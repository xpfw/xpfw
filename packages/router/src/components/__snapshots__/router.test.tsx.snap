// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Fields Test: back home 1`] = `
<div
  class="home"
>
  {"route":{"path":"/","params":{}}}
</div>
`;

exports[`Basic Fields Test: back to b 1`] = `
<div
  class="b"
>
  {"route":{"path":"/b","params":{"my":"arg"}}}
</div>
`;

exports[`Basic Fields Test: back to c 1`] = `
<div
  class="c"
>
  {"route":{"path":"/c","params":{"gfsdf":"tas"}}}
</div>
`;

exports[`Basic Fields Test: back to home 1`] = `
<div
  class="home"
>
  {"route":{"path":"/","params":{}}}
</div>
`;

exports[`Basic Fields Test: back to home multi back no effect 1`] = `
<div
  class="home"
>
  {"route":{"path":"/","params":{}}}
</div>
`;

exports[`Basic Fields Test: invalid oute 1`] = `
<div
  class="404"
>
  {"route":{"path":"/gfadsdf","params":{"jk":"hj"}}}
</div>
`;

exports[`Basic Fields Test: renderig 404 1`] = `
<div
  class="404"
>
  {"route":{"path":"/","params":{}}}
</div>
`;

exports[`Basic Fields Test: renderig a 1`] = `
<div
  class="a"
>
  {"route":{"path":"/a","params":{"xvcb":"asdf"}}}
</div>
`;

exports[`Basic Fields Test: renderig a different params 1`] = `
<div
  class="a"
>
  {"route":{"path":"/a","params":{"gewar":"yvcxv"}}}
</div>
`;

exports[`Basic Fields Test: renderig a with old params because back 1`] = `
<div
  class="a"
>
  {"route":{"path":"/a","params":{"xvcb":"asdf"}}}
</div>
`;

exports[`Basic Fields Test: renderig b 1`] = `
<div
  class="b"
>
  {"route":{"path":"/b","params":{"my":"arg"}}}
</div>
`;

exports[`Basic Fields Test: renderig c 1`] = `
<div
  class="c"
>
  {"route":{"path":"/c","params":{"gfsdf":"tas"}}}
</div>
`;

exports[`Basic Fields Test: still invalid oute 1`] = `
<div
  class="404"
>
  {"route":{"path":"/gfadsdf","params":{"iou":".-รถ"}}}
</div>
`;
